rpc/server_interceptors.go

```diff
@@ -0,0 +1,126 @@
+package rpc
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// UnaryServerTracingInterceptor starts a new Span if Span metadata exists in the context.
+func UnaryServerTracingInterceptor(logger golog.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		remoteSpanContext, err := remoteSpanContextFromContext(ctx)
+		var span *trace.Span
+		if err == nil {
+			ctx, span = trace.StartSpanWithRemoteParent(ctx, "server_root", remoteSpanContext)
+			defer span.End()
+		} else {
+			logger.Warnf("client did not send valid Span metadata in headers, local Spans will not be linked to client. reason: %w", err)
```

Comment:
This should be a key-value pair instead of using `%w`. See https://github.com/viamrobotics/goutils/blob/main/jwks/jwksutils/utils.go#L82